### PR TITLE
Add support for Jenkins Pipeline jobs

### PR DESCRIPTION
**Pull request by cprice404**
With these commits, the plugin can now be used from Jenkins Pipeline jobs, by adding a line like this to your pipeline build script:

    gatlingArchive()

### CODE REVIEW

**Comment by cprice404** [comment]
Fixes gatling/gatling#3018

File: src/main/resources/io/gatling/jenkins/GatlingBuildAction/summary.jelly [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+        <b>${%AvailableReports} :</b>
+        <ul>
+            <j:forEach items="${it.simulations}" var="sim">
+                <li>
+                    <a href="${it.getReportURL(sim.simulationName)}">${sim.simulationName}</a>
+                </li>
+            </j:forEach>
+        </ul>
+    </t:summary>
+</j:jelly>
```
Comment (slandelle) [inline src/main/resources/io/gatling/jenkins/GatlingBuildAction/summary.jelly]: missing empty line?

File: src/main/resources/io/gatling/jenkins/GatlingBuildAction/report.properties [hunk 2]
```diff
@@ -1 +1 @@
-OpenNewPage=Open in a new page.
\ No newline at end of file
+OpenNewPage=Open in a new window
```
Comment (slandelle) [inline src/main/resources/io/gatling/jenkins/GatlingBuildAction/report.properties]: missing empty line?

File: src/main/java/io/gatling/jenkins/GatlingPublisher.java [hunk 3]
```diff
@@ -155,6 +156,7 @@ public Action getProjectAction(AbstractProject<?, ?> project) {
     return reportsFromThisBuild;
   }
 
+
```
Comment (slandelle) [inline src/main/java/io/gatling/jenkins/GatlingPublisher.java]: kill extra empty line

**Comment by slandelle** [comment]
Awesome Chris, thanks a lot! We'll give it a try asap and release a new version.
Where did you find a migration documentation?

**Comment by cprice404** [comment]
Pushed up a commit with the proposed whitespace fixes.

@slandelle I didn't really find any comprehensive docs on migration; it was kind of a combination of asking questions in IRC and on the mailing list, looking at the source code of other plugins and of Jenkins itself, and just trying things out until I understood it.  The closest thing to docs that I found was this:

https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md

That actually has most of the information, it's just written in a way that is pretty hard to understand until you get into the code.  This also has some useful information:

https://github.com/jenkinsci/workflow-step-api-plugin

I was planning on writing some kind of blog post / tutorial about the upgrade experience to share with Jenkins and hopefully make things easier for the next person :)

**Comment by slandelle** [comment]
not_bad

**Comment by slandelle** [comment]
@cprice404 Could you please squash your commits into a single one?

**Comment by cprice404** [comment]
squashed.

**Comment by slandelle** [comment]
Awesome work, thanks!

**Comment by jglick** [comment]
Does not make sense to _both_ make `GatlingPublisher` implement `SimpleBuildStep` _and_ create a dedicated `Step`. You can do the first if you have no special needs beyond “make what worked in a freestyle project also be available in a Pipeline build”; the second if you wanted to customize the options and behaviors available from Pipeline.

**Comment by cprice404** [comment]
@jglick is it harmful to support both?  e.g. you can trigger the 'archive' functionality either way (though I recgonize that the code paths are slightly different).

**Pull request merged**